packages/web-components/package.json

```diff
@@ -28,7 +28,7 @@
   },
   "dependencies": {
     "@department-of-veterans-affairs/css-library": "^0.8.1",
-    "@stencil/core": "^3.2.1",
+    "@stencil/core": "4.0.0",
```

Comment:
I don't want to open pandora's box but did you happen to try to latest release of v4? It looks like they're up to v4.21 and [v4.0.0 is not actively maintained](https://stenciljs.com/docs/v4.0/introduction).

Reply:
The later versions have issues with the piercing (`>>>`) selector in the e2e tests. I have checked the documentation and the how-to pages and they all say that this should still work in 4.21 but they don't. This is the error I get when running tests using 4.21:

```
Evaluation failed: DOMException: Failed to execute 'querySelector' on 'Document': 'va-select >>> select' is not a valid selector.
```

Reply:
Interesting. I guess that makes wonder a few more things:

1. Are we ok just stopping here for now (like what does being on `v4.0.0` unblock for us now?) or is stopping here just delay us eventually needing to solve for it?
2. Should we create a follow-up ticket to get us onto the latest (or a more recent) version?

Glancing at their [release notes](https://github.com/ionic-team/stencil/releases), version [4.14.0](https://github.com/ionic-team/stencil/releases/tag/v4.14.0) seems particularly useful for testing because they added support for [deep piercing with Puppeteer](https://github.com/ionic-team/stencil/pull/5481). I know we've had some problems dealing with this in our tests and we had testing gaps because of not being able to do that.

There are probably other useful things too I just haven't really dug into what else they've added.

Reply:
Here is an example where we could use deep piercing but instead had to skip the test because of flakiness:

https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-file-input/test/va-file-input.e2e.ts#L89-L95

Reply:
I've been trying to get this to work locally but haven't succeeded yet. Trying to see what happens when jest is running on the latest version in addition to stencil 4.21.0. Currently getting some weird errors :\

![image](https://github.com/user-attachments/assets/610a3c80-cf88-4683-9ece-85371e43be8a)